docs/api-reference/hooks.md

```diff
@@ -0,0 +1,162 @@
+# Hooks
+
+## useNavigation
+
+This hook exposes all the navigating capabilities of Yarr. 
+
+```jsx
+import {useNavigation} from 'yarr'
+
+// in a hook/component
+const navigation = useNavigation();
+```
+
+The `navigation` object contains the following methods
+
+| Method      | Description |
+| ----------- | ----------- |
+| push     | Pushes a new history entry on to the history stack and navigates to that route           |
+| go(n: number)    | Moves the history stack pointer by n entries            |
+| replace     |  Replaces the current history entry with a new location and navigates          |
+| block     | Blocks all navigation until unblocked.            |
+| back     |  Goes back to the previous location if it exists. Equivalent to `go(-1)`          |
+| forward     |  Goes to the next location if it exists. Equivalent to `go(1)`           |
+--- 
+
+While the `Link` component is very convenient to use, it only renders an `anchor` element. This hook `useRouteProps` can make navigation easy with any component. For eg, 
+
+### Navigation when a button is clicked
+
+```jsx
+const navigation = useNavigation()
+const goToAboutPage = () => {
+    navigation.push(`/about`);
+}
+
+<button onClick={goToAboutPage}> Go to About Page </button>
+```
+
+### History entries with more information
+
+Besides navigation using string location descriptors like `/about`. Methods like `push` and `replace` also accept objects to describe locations. 
+
+- The location descriptors can include the `pathname`, `search` query property and the `state` property within the object in the transition. 
+
+
+For eg: 
+
+```
+navigation.push({
+    pathname: `/about`,
+    search: `?a=5&b=6`,
+    state: {
+      firstVisitToPage: 1  
+    }
+})
+```
+
+
+## useRouteProps
+`useRouteProps` , a very commonly used hook, can be used to access information related to the current active route. 
+
+### Usage
+
+```jsx
+const {params, search, preloaded} = useRouteProps();
+```
+---
+
+### Params example
+
+Let us consider an app which has a user profile route `/user/:username`. 
+
+Now when the user is on the URL `/user/lee`, then 
+
+```jsx
+const {params} = useRouteProps();
+console.log(params.username) // lee
```

Comment:
**NOTE**
.username here could be `undefined`.

It's important to note that usage of `useRouteProps()` doesn't guarantee the params are available because the usage could be on _any_ route.

Using params passed to the page components is a guarenteed way of getting param props.